libs/net/core/Exceptions/HttpClientRequestException.cs

```diff
@@ -49,7 +49,6 @@ public HttpClientRequestException(string message, Exception innerException, Http
         public HttpClientRequestException(HttpResponseMessage response) : base($"HTTP Request failed [{response?.RequestMessage?.Method}]:[{response?.StatusCode}]:{response?.RequestMessage?.RequestUri}", null, response?.StatusCode)
         {
             this.Response = response ?? throw new ArgumentNullException(nameof(response));
-            this.Data["Body"] = response.Content.ReadAsStringAsync().Result;
```

Comment:
Don't read the body here.  Let the handler decide how to handle the exception.  A body can only be read once.